public/angular-app/settings/settings-controller.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Created by konrad.sobon on 2019-05-02.
+ */
+angular.module('MissionControlApp').controller('SettingsController', SettingsController);
+
+function SettingsController(SettingsFactory, ngToast, $route){
+    var vm = this;
+    var toasts = [];
+    vm.settings = null;
+    vm.office = { name: null, code: null };
+
+    getSettings();
+
+    /**
+     * Adds Office Location to the list.
+     * @param arr
+     * @constructor
+     */
+    vm.AddOffice = function (arr) {
+        if(vm.office.name === null || vm.office.code === null) return;
```

Comment:
In order to handle things like empty strings (if someone typed then deleted), you might want to do something like:
`if(!vm.office.name || !vm.office.code) return;` 

Which coerces `null`, `undefined`, and an empty string to false.

Reply:
It may also be good to use `trim` on the two fields to remove whitespace at the beginning and end, just to be safe.

Reply:
Fixed.